citest/google_session/google_session.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import requests
+
+from bs4 import BeautifulSoup
+
+class GoogleSession:
+  """Utility class for automating Google authentication.
+
+  Automates both the Google account login and scope approval to create an
```

Comment:
This class is intended to automate authenticating with Google as an identity provider in Spinnaker. It's intended to be agnostic of what the actual protocol is behind the scenes.

The general flow in the supported protocols are:

1. Try to access an application that needs access to a resource
2. Get redirected to login page for your identity provider
3. Authenticate and grant permission to the application for the resource it needs
4. Some protocol-specific redirects with various state exchanged
5. Wind up back at the application with the correct permissions granted, so the application can behave correctly

This class automates (3) so that we can run integration tests without a user present and verify that the components that act differently given an authenticated user behave properly. Should I put this in the description of the module so that the intent is clearer?